ACE/ace/Log_Msg.cpp

```diff
@@ -1791,16 +1791,25 @@ ACE_Log_Msg::log (const ACE_TCHAR *format_str,
                       ACE_OS::sprintf (bp,
                                        format,
                                        static_cast <unsigned> (ACE_Thread::self ()));
-#elif defined ACE_USES_WCHAR
+#else
+
+#  ifdef ACE_HAS_GETTID
+#    define ACE_LOG_MSG_GET_THREAD_ID ACE_OS::thr_gettid
+#  else
+#    define ACE_LOG_MSG_GET_THREAD_ID ACE_OS::thr_id
+#  endif
+
+#  if defined ACE_USES_WCHAR
                   {
                     char tid_buf[32] = {};
```

Comment:
When ACE_HAS_GETTID is defined, do we need a buffer of size 32 or can we use a smaller buffer?

Reply:
tids are drawn from the same pool as pids. It seems the maximum possible limit on that pool for 64bit systems is 2^22 or 4194304, 5 digits. So yes it could be smaller if 24 bytes (assuming we make it 8) is worth shaving off the stack.

Reply:
> 4194304, 5 digits

Apparently I forgot how to count. I meant to say 7.